### ISSUE TITLE
Inaccurate predictions because of delta_time

### ISSUE THREAD

**Issue by Darkluke1111** [Bendzae/bevy_replicon_snap#12]
Using the `delta_time` parameter while implementing the `apply_event` function for the `Predict` trait causes inaccurate predictions if the update rate of the server and the client are different.

**Comment by Bendzae** [comment]
Yes, I think the prediction code in general needs some love. Will try to improve that once I find the time.

**Comment by Darkluke1111** [comment]
I'm no networking expert but I feel like in cases similar to the included box example the way to go would be to include the client's delta_time for the move event and send it alongside the direction to the server. Then the server can use that time to calculate the target position independent of its own update time.
This means that the detla_time parameter in the `apply_event` function isn't required, since that time is taken from the event struct.
Of course, letting clients send those times without any serverside checking lets clients cheat by just setting them higher. My first thought is that to prevent this, maybe clients should send timestamps instead of deltas so the server can check those to be valid and then calculate the deltas from them.
I will do some research on how to do this properly and maybe make a PR if I find a good solution.

**Comment by Darkluke1111** [comment]
So after giving this a bit more thought, I think the main problem here is to send and apply client events in the regular update schedule instead of the fixed-update schedule. In a server-authoritative game, client events will in 99% of the cases be input events (or events that are closely coupled to user inputs) and those are best handled in the fixed-update schedule ([Fixed Timestep - Unofficial Bevy Cheat Book](https://bevy-cheatbook.github.io/fundamentals/fixed-timestep.html)). So the easiest way would probably be to run the whole prediction logic in the fixed update schedule. Then the whole miss prediction problem could be solved by just making sure the server and the client have the same fixed update timestep.
The [examples from the lightyear crate](https://github.com/cBournhonesque/lightyear/tree/main/examples/simple_box) also suggest that this is probably the way to go.
